src/librustc_typeck/diagnostics.rs

```diff
@@ -12,6 +12,28 @@
 
 register_long_diagnostics! {
 
+E0046: r##"
+When trying to make some type implement a trait `Foo`, you must, at minimum,
+provide implementations for all of `Foo`'s required methods (meaning the
+methods that do not have default implementations), as well as any required
+trait items like associated types or constants.
+"##,
+
+E0054: r##"
+It is not allowed to cast to a bool. If you are trying to cast a numeric type
```

Comment:
Perhaps this could clarify "It is not allowed to cast a numeric type to a bool"?